src/ansys/fluent/core/launcher/pim_launcher.py

```diff
@@ -245,6 +254,30 @@ def launch_remote_fluent(
 
     instance.wait_for_ready()
 
+    file_service = PimFileTransferService(pim_instance=instance)
+
+    local_directory = os.path.join(os.getcwd(), f"sifile_{uuid.uuid4().hex}")
+
+    max_retries = 20
+    wait_time_between_retries = 1  # seconds
+
+    for i in range(max_retries):
+        try:
+            file_service.download_file("sifile.txt", local_directory)
+            break
+        except Exception as ex:
+            if i == max_retries - 1:
+                raise TimeoutError(
+                    "Failed to download file after multiple retries."
```

Comment:
Since we know how the retries were configured at this point, we can (easily) add that information in the message for full transparency.